### PR TITLE
Added Security in the API

### PR DESCRIPTION
**Pull request by yasharth291**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -25,6 +25,18 @@ const jwt = require("jsonwebtoken");
...
+const rateLimit = require("express-rate-limit");
+const xss  = require("xss-clean");
+const helmet = require("helmet");
+const mongoSanitize = require("express-mongo-sanitize");
+
+const apiLimiter = rateLimit({
+  windowMs: 60 * 60 * 1000, // 15 minutes
+  max: 500,// this can be edited in between
+  message:
+  "Too many accounts created from this IP, please try again after an hour"
```
Comment (iamsaumya) [inline index.js:37]: Could you please change this to "Too many requests from this IP, please try again after an hour"?

File: package.json [hunk 2]
```diff
@@ -28,15 +28,19 @@
...
     "image-hash": "^4.0.1",
     "jest": "^26.0.1",
     "jsonwebtoken": "^8.5.1",
     "moment": "^2.27.0",
     "mongoose": "^5.11.20",
     "nodemon": "^2.0.4",
     "shortid": "^2.2.15",
-    "validator": "^13.0.0"
+    "validator": "^13.0.0",
+    "xss-clean": "^0.1.1"
```
Comment (iamsaumya) [inline package.json:43]: Could you please choose another package? This package was last published 5 years ago and the author himself is not using this anymore [here](https://github.com/jsonmaur/xss-clean/issues/1#issuecomment-701678780)

**Review by iamsaumya — APPROVED** [review]
Hey Yasharth, this is a great improvement.

**Comment by yasharth291** [comment]
sure

**Comment by iamsaumya** [comment]
@yasharth291 we have a lot of Travis-ci checks failing, could you please update and add appropriate tests as well?

**Comment by iamsaumya** [comment]
Also, I realised that this PR is made without getting a green signal from the maintainers. A general rule of thumb for an open-source project is to discuss the changes with the maintainer in the ISSUE and then create a PR if the maintainers approve. 

Otherwise, the maintainers have to read and review a lot of PR's which might not be merged and is unhealthy for the project.

**Comment by yasharth291** [comment]
Screenshot_20210312-153923
please do comment before having adequate knowledge about the matter.

**Comment by yasharth291** [comment]
as he was not able to talk in the issue. I discussed about the PR in slack group.

**Comment by iamsaumya** [comment]
Apologies if I have offended you in any way.

However, it is in the spirit of open source to talk about the issue in public so that other people can keep track and extend help if necessary.

That being said, it is the choice of the maintainers to choose the medium to communicate and I respect that. 

Thank you!

**Comment by yasharth291** [comment]
no problem 👍☺️☺️ ☮️

**Comment by iamsaumya** [comment]
btw if Palisadoes Foundation has a Slack channel, I want to hope in as well. Could you please tell me where can I find the invite link?

**Comment by yasharth291** [comment]
you have to contact the maintainer for it mail them they will send

**Comment by yasharth291** [comment]
thats how i got it

**Comment by Laurell876** [comment]
@iamsaumya I just added you to the palisadoes foundation's slack channel. Also the travis ci checks are failing because the hosted version of the api is down, I hope to get it fixed later today.

**Pull request merged**